### PR TITLE
hwloc: replace 'shared' variant with 'libs'

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by bgoglin** [comment]
I don't see any issue but I am not familiar enough with spack to review with some confidence.

File: var/spack/repos/builtin/packages/hwloc/package.py [hunk 1]
```diff
@@ -74,7 +74,13 @@ class Hwloc(AutotoolsPackage):
         default=(sys.platform != "darwin"),
         description="Support analyzing devices on PCI bus",
     )
-    variant("shared", default=True, description="Build shared libraries")
+    variant(
+        "libs",
+        default="shared,static",
+        values=("shared", "static"),
+        multi=True,
+        description="Build shared libs, static libs or both",
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hwloc/package.py:82]: Is the ability to build both libraries supported in all versions listed in this package or only from some version on?

**Review by tldahlgren — APPROVED** [review]
LGTM.  Just have the question about multi-library support.

File: var/spack/repos/builtin/packages/hwloc/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/hwloc/package.py:82]: As far as I can tell, for all versions.

Comment (haampie) [reply var/spack/repos/builtin/packages/hwloc/package.py:82]: It's more of an autotools thing, the pattern is quite common now in other packages. I'm fine with this change. Whenver a variant changes name, it's good to verify that it's not assumed to exist in things such as `setup_run_environment` etc. In this case not, so it's not a terribly breaking change to rename the variant.

**Review by haampie — APPROVED** [review]

**Pull request merged**